src/test/java/org/fsg1/fmms/backend/services/CurriculumServiceTest.java

```diff
@@ -0,0 +1,94 @@
+package org.fsg1.fmms.backend.services;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.fsg1.fmms.backend.database.Connection;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static junit.framework.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+
+public class CurriculumServiceTest {
+    final String query = "SELECT coalesce(array_to_json(array_agg(row_to_json(co))), '[]'::json) as semesters\n" +
```

Comment:
Hardcoded query string? Is this really necessary?

Reply:
Yes, it's to verify the method was called with this string.

`verify(conn, times(1)).executeQuery(query, "SE");`

Reply:
I mean, it is in the code twice now. Can't you  load it from the class. Via package private static variable?

Reply:
I see what you mean I'll fix it.